lib/App/Config/Chronicle.pm

```diff
@@ -347,6 +350,316 @@ sub current_revision {
+    $self->chronicle_writer->mset(\@atomic_write_pairs, $date_obj, @$optional_chron_args);
+    return 1;
+}
+
+=head2 get
+
+Takes either an arrayref of keys, gets them atomically, and returns a hashref of key->values,
+including the global_revision under the key '_global_revision'
+If a key has an empty value, it will return with undef.
+
+For convenience a get with just a key string will return the value only.
+
+Example:
+    get(['key1','key2','key3',...]);
+Returns:
+    {'key1' => 'value1', 'key2' => 'value2', 'key3' => 'value3',..., '_global_revision' => '<number>'}
+
+Example:
+    get(['key1']);
+Returns:
+    {'key1' => 'value1', '_global_revision' => '<number>'}
+
+Example:
+    get('key1');
+Returns:
+    'value1'
+
+=cut
+
+sub get {
+    my ($self, $keys) = @_;
+    my $single_get;
+
+    if (ref $keys eq '') {
+        $keys = [$keys] if ref $keys eq '';
+        $single_get = 1;
+    }
+    grep { die "Cannot get with key: $_ | Key must be defined'" unless $self->_key_exists($_) } @$keys;
+
+    push @$keys, '_global_rev' unless $single_get;
+
+    my @result_objs = $self->_retrieve_objects($keys);
+
+    return $result_objs[0] ? $result_objs[0]->{data} : $self->get_default($keys->[0]) if $single_get;
+    return {map { $keys->[$_] => $result_objs[$_] ? $result_objs[$_]->{data} : $self->get_default($keys->[$_]) } (0 .. scalar @$keys - 1)}
+        unless $single_get;
+
+}
+
+sub _retrieve_objects {
+    my ($self, $keys) = @_;
+    return $self->_retrieve_objects_from_cache($keys) if $self->local_caching;
+    return $self->_retrieve_objects_from_chron($keys);
+}
+
+sub _retrieve_objects_from_cache {
+    my ($self, $keys) = @_;
+    return map { $self->{$_} } @$keys;
+}
+
+sub _retrieve_objects_from_chron {
+    my ($self, $keys) = @_;
+    my @atomic_read_pairs = map { [$self->setting_namespace, $_] } @$keys;
+    return $self->chronicle_reader->mget(\@atomic_read_pairs);
+}
+
+=head2 get_history
+
+Retreives a past revision of an app config entry, where $rev is the number of revisions in the past requested.
+If the third argument is set to 1 the result of the query will be cached in Redis. This is useful if a certain
+    revision will be needed repeatedly, to avoid excessive database access. By default this argument is 0.
+All cached revisions will become stale if the key is set with a new value.
+
+Example:
+    get_history('system.email', 0); Retrieves current version
+    get_history('system.email', 1); Retreives previous revision
+    get_history('system.email', 2); Retreives version before previous
+
+=cut
+
+sub get_history {
+    my ($self, $key, $rev, $cache) = @_;
+    $cache //= 0;
+
+    die "Cannot get history of key: $key | Key must be dynamic" unless $self->_key_is_dynamic($key);
+
+    my @data_objs = $self->_retrieve_objects([$key, $key . '::' . $rev]);
+    my $curr_obj  = $data_objs[0];
+    my $hist_obj  = $data_objs[1];
```

Comment:
You could do this in one line with a list-unpacking assignment:

    my ($hist_obj, $curr_obj) = $self->_retrieve_objects ...